Lambdas/GHHooks/Models/+SemanticVersion.swift

```diff
@@ -0,0 +1,111 @@
+import SwiftSemver
+
+enum SemVerBump {
+    case releaseStage
+    case major
+    case minor
+    case patch
+}
+
+extension SemanticVersion {
+    func next(_ bump: SemVerBump) -> SemanticVersion? {
+        var version = self
+
+        if version.prereleaseIdentifiers.isEmpty {
+            switch bump {
+            case .releaseStage:
+                return nil
+            case .patch:
+                version.patch += 1
+            case .minor:
+                version.patch = 0
+                version.minor += 1
+            case .major:
+                version.patch = 0
+                version.minor = 0
+                version.major += 1
+            }
+        } else {
+            guard version.prereleaseIdentifiers.count < 4 else {
+                /// Shouldn't have more than 3 identifiers, otherwise unexpected by this code.
+                return nil
+            }
+            /// At this point we are guaranteed 1/2/3 `prereleaseIdentifiers`.
+
+            guard !version.prereleaseIdentifiers[0].allSatisfy(\.isNumber) else {
+                /// Identifiers should be like `["alpha", "1"]`.
+                /// First identifier should not be a number.
+                return nil
+            }
+
+            guard version.prereleaseIdentifiers[1...]
+                .allSatisfy({ $0.allSatisfy(\.isNumber) }) else {
```

Comment:
```suggestion
            guard version.prereleaseIdentifiers[1...].allSatisfy({ UInt($0) != nil }) else {
```
(The "allSatisfy is number" check would accept absurdly long numbers and empty strings, for example.)